packages/cohort_banking/src/callbacks/statemap_installer.rs

```diff
@@ -1,68 +1,106 @@
 use std::{sync::Arc, time::Duration};
 
 use async_trait::async_trait;
-use talos_cohort_replicator::{ReplicatorInstallStatus, ReplicatorInstaller, StatemapItem};
+use deadpool_postgres::Transaction;
+use talos_cohort_replicator::{ReplicatorInstaller, StatemapItem};
 use tokio_postgres::types::ToSql;
 
 use crate::{model::requests::TransferRequest, state::postgres::database::Database};
 
-pub struct StatemapInstallerImpl {
+const BANK_ACCOUNTS_UPDATE_QUERY: &str = r#"
+UPDATE bank_accounts ba SET
+"amount" =
+(CASE
+    WHEN ba."number" = ($1)::TEXT THEN ba."amount" + ($3)::DECIMAL
+    WHEN ba."number" = ($2)::TEXT THEN ba."amount" - ($3)::DECIMAL
+    END),
+    "version" = ($4)::BIGINT
+    WHERE ba."number" IN (($1)::TEXT, ($2)::TEXT)
+    AND ba."version" < ($4)::BIGINT
+    "#;
+
+const SNAPSHOT_UPDATE_QUERY: &str = r#"UPDATE cohort_snapshot SET "version" = ($1)::BIGINT WHERE id = $2 AND "version" < ($1)::BIGINT"#;
+
+fn is_retriable_error(error: &str) -> bool {
+    // TODO: improve retriable error detection when error moves from String to enum or struct.
```

Comment:
Agree, this is Postgres text. Somehow it should be abstracted depending on DB type.